Azure/ARO-RP
# upgrade go version to 1.22

### Which issue this PR addresses:

Fixes: https://issues.redhat.com/browse/ARO-9866

### What this PR does / why we need it:
 This PR upgrades go version from `1.21` to `1.22` and pulls the Golang container image `mcr.microsoft.com/oss/go/microsoft/golang:1.22.6-fips-cbl-mariner2.0` from Microsoft container registry.

### Test plan for issue:
1. Ensure E2E is success.
2. Build and Deploy ARO-RP image with new go version in INT.

### Is there any documentation that needs to be updated for this PR?
No

### How do you know this will function as expected in production? 
- Testing successfully in INT
